frontend/src/pages/Settings.tsx

```diff
@@ -1,22 +1,59 @@
-import  { useState, useEffect } from 'react';
-import { Version } from "../../wailsjs/go/main/App";
+import { useState, useEffect, useContext, useMemo } from 'react';
+import { UpdateSettings, Version } from '../../wailsjs/go/main/App';
 
 import { Heading, Subheading } from '../components/catalyst/heading'
 import { Divider } from '../components/catalyst/divider'
 import { Text } from '../components/catalyst/text'
+import { Field, FieldGroup, Label } from '../components/catalyst/fieldset';
+import { Input } from '../components/catalyst/input';
+import { Button } from '../components/catalyst/button';
+import { SettingsContext } from '../App';
 
 export function Settings() {
+    const context = useContext(SettingsContext);
+    if (!context) throw new Error("SettingsContext must be used within a SettingsProvider");
+    const { settings } = context;
+    const { refreshSettings } = useContext(SettingsContext)!
     const [version, setVersion] = useState<string>("");
+    const [openaiKey, setOpenaiKey] = useState<string>(settings?.openai_key || '');
+    const [error, setError] = useState('')
 
     useEffect(() => {
         Version().then(setVersion);
     }, []);
 
+    useEffect(() => {
+        setOpenaiKey(settings?.openai_key || '');
+    }, [settings]);
+
+    const handleSaveSettings = () => {
+        // Trim the openaiKey only when saving
+        const trimmedKey = openaiKey.trim()
+
+        // Create a new Settings object with trimmed openai_key
+        const data = {
+            openai_key: trimmedKey,
+        };
+
+        // @ts-ignore
+        UpdateSettings(data)
+            .then(() => {
+                // Refresh the settings from the database
+                refreshSettings();
+            })
+            .catch((err: any) => {
+                console.error('Error saving credentials:', err);
+                setError('Failed to save credentials');
+            });
+    };
+
+    const dirty = useMemo(() => openaiKey !== settings?.openai_key, [openaiKey, settings])
```

Comment:
maybe it's simpler to have a flag "has changed" and whenever the user changes a field (even if it is to set the same value) we allow them to "save".

That should simplify the logic when we add more fields